packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -1059,74 +1175,140 @@ class _SelectToggleButtonRenderObject extends RenderShiftedBox {
     if (child == null) {
-      return constraints.constrain(Size(
-        leadingBorderSide.width + trailingBorderSide.width,
-        horizontalBorderSide.width * 2.0,
-      ));
+      if (direction == Axis.horizontal) {
+        return constraints.constrain(Size(
+          leadingBorderSide.width + trailingBorderSide.width,
+          borderSide.width * 2.0,
+        ));
+      } else {
+        size = constraints.constrain(Size(
+          borderSide.width * 2.0,
+          leadingBorderSide.width + trailingBorderSide.width,
+        ));
+      }
     }
 
     final double leftConstraint;
     final double rightConstraint;
+    final double topConstraint;
+    final double bottomConstraint;
+
+    if (direction == Axis.horizontal) {
+      switch (textDirection) {
+        case TextDirection.ltr:
+          rightConstraint = _trailingBorderOffset;
+          leftConstraint = leadingBorderSide.width;
+          topConstraint = borderSide.width;
+          bottomConstraint = borderSide.width;
+
+          final BoxConstraints innerConstraints = constraints.deflate(
+            EdgeInsets.only(
+              left: leftConstraint,
+              top: topConstraint,
+              right: rightConstraint,
+              bottom: bottomConstraint,
+            ),
+          );
 
-    switch (textDirection) {
-      case TextDirection.ltr:
-        rightConstraint = _trailingBorderOffset;
-        leftConstraint = leadingBorderSide.width;
-
-        final BoxConstraints innerConstraints = constraints.deflate(
-          EdgeInsets.only(
-            left: leftConstraint,
-            top: horizontalBorderSide.width,
-            right: rightConstraint,
-            bottom: horizontalBorderSide.width,
-          ),
-        );
+          final Size childSize = layoutChild(child!, innerConstraints);
+
+          return constraints.constrain(Size(
+            leftConstraint + childSize.width + rightConstraint,
+            borderSide.width * 2.0 + childSize.height,
+          ));
+        case TextDirection.rtl:
+          rightConstraint = leadingBorderSide.width;
+          leftConstraint = _trailingBorderOffset;
+          topConstraint = borderSide.width;
+          bottomConstraint = borderSide.width;
+
+          final BoxConstraints innerConstraints = constraints.deflate(
+            EdgeInsets.only(
+              left: leftConstraint,
+              top: topConstraint,
+              right: rightConstraint,
+              bottom: bottomConstraint,
+            ),
+          );
 
-        final Size childSize = layoutChild(child!, innerConstraints);
+          final Size childSize = layoutChild(child!, innerConstraints);
+          return constraints.constrain(Size(
+            leftConstraint + childSize.width + rightConstraint,
+            borderSide.width * 2.0 + childSize.height,
+          ));
+      }
+    } else {
+      final double trailingBorderOffset = isLastButton ? trailingBorderSide.width : 0.0;
```

Comment:
This should be in the `VerticalDirection.down` case for the below switch statement

Reply:
I rewrote `_computeSize`, now it looks more clean.